numba_cuda/numba/cuda/cpython/numbers.py

```diff
@@ -0,0 +1,1471 @@
+import math
+import numbers
+
+import numpy as np
+import operator
+
+from llvmlite import ir
+from llvmlite.ir import Constant
+
+from numba.core.imputils import (
```

Comment:
By importing and using `lower_*` from Numba core, we are re-registering duplicate implementations in the core of Numba. I think we should be using our own registry (that we can create in this module) here.